File: mockez-back/src/main/java/com/mockez/service/SchemaService.java
package com.mockez.service;

import com.mockez.domain.model.entity.Schema;

import java.util.List;
import java.util.UUID;

/**
 * @author Luong Quoc Trung, Do Quoc Viet
 */

public interface SchemaService {

    /**
     * SCHEMA.01 GET /schema/{projectId}
     * @param projectId of project
     * @return schemas by projectId
     */
    List<Schema> getSchemasByProject(UUID projectId);

    UUID saveOrUpdate(Schema schema);

    UUID delete(UUID id);
}

File: mockez-front/src/app/core/service/schema.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Schema } from '@core/model/schema';
import { environment } from '@environment/environment';

/**
 * @author Luong Quoc Trung, Do Quoc Viet
 */

@Injectable({
  providedIn: 'root'
})
export class SchemaService {

  constructor(
    private httpClient: HttpClient
  ) {
  }

  getSchemasByProject(projectId: string): Observable<Schema[]> {
    return this.httpClient.get<Schema[]>(`${environment.API_URL}/schema?projectId=${projectId}`);
  }

  saveOrUpdate(schema: Schema): Observable<string> {
    return this.httpClient.post<string>(`${environment.API_URL}/schema`, schema);
  }

  deleteSchema(id: string): Observable<string> {
    return this.httpClient.delete<string>(`${environment.API_URL}/schema/${id}`);
  }

  getSchema(id: string): Observable<Schema> {
    return this.httpClient.get<Schema>(`${environment.API_URL}/schema/${id}`);
  }
}
